lib/hooks/puzzle.ts

```diff
@@ -0,0 +1,56 @@
+import useSWR from "swr";
+import IDailyPuzzle from "../../types/IDailyPuzzle";
+
+const fetcher = (url: string) => fetch(url).then((r) => r.json());
+
+/**
+ * Gets today's daily puzzle.
+ */
+export const useDailyPuzzle = () => {
+  const { data, error } = useSWR<IDailyPuzzle>("/api/puzzle/daily", fetcher);
```

Comment:
I personally do not like the way that the words for the daily puzzle is chosen -- how the engineer has to manually choose a word and enter the "daysAgo." I think it would be better to have a word bank and select words randomly. This gives each word an equal probability and bars any bias from the engineer, including convenience. information, and recency bias.

Reply:
This hooks file is on the frontend and simply fetches from the backend, it does not directly interact with the database.
The reason a document was created for each puzzle was to allow the daily statistics to be tied to the puzzle document, allowing this database document to contain the leaderboard as well as counts of attempts and wins.
Eventually a dashboard or importer feature can be added to automate the adding of words from a word bank.